assets/js/phoenix.js

```diff
@@ -967,6 +968,11 @@ export class Socket {
    */
   remove(channel){
     this.channels = this.channels.filter(c => c.joinRef() !== channel.joinRef())
+
+    this.stateChangeCallbacks.open = this.stateChangeCallbacks.open.filter(entry => {
+      const topic = Object.keys(entry)[0]
+      return topic !== channel.topic
+    })
```

Comment:
instead of stateChangeCallbacks filter, we can simply call `off` before our existing channels filter:

```javascript
  remove(channel){
    this.off(channel.topic)
    this.channels = this.channels.filter(c => c.joinRef() !== channel.joinRef())
  }
```